etrago/tools/plot.py

```diff
@@ -2293,85 +2398,180 @@ def plot_grid(
         all_network, line_colors, link_colors = calc_network_expansion(
             network, method="abs", ext_min=ext_min
         )
         plot_background_grid(all_network, ax)
-        if ext_width != False:
-            line_widths = 0.5 + (line_colors / ext_width)
-            link_widths = 0.5 + (link_colors / ext_width)
+
+        if ext_width is not False:
+            line_widths = line_colors / ext_width
+            link_widths = link_colors.apply(
+                lambda x: x / ext_width if x != 0 else 0
+            )
+        else:
+            dc_link = network.links.index[network.links.carrier == "DC"]
+            link_widths = pd.Series(0, index=network.links.index)
+            link_widths.loc[dc_link] = 1.5
+            line_widths = line_colors.apply(lambda x: 1.5 if x != 0 else 0)
+
+        link_colors = link_colors.mul(1e-3)
+        line_colors = line_colors.mul(1e-3)
+
     elif line_colors == "expansion_rel":
         title = "Network expansion"
         label = "network expansion in %"
         all_network, line_colors, link_colors = calc_network_expansion(
             network, method="rel", ext_min=ext_min
         )
         plot_background_grid(all_network, ax)
-        if ext_width != False:
+        if ext_width is not False:
             line_widths = 0.5 + (line_colors / ext_width)
-            link_widths = 0.5 + (link_colors / ext_width)
+            link_widths = link_colors.apply(
+                lambda x: 0.5 + x / ext_width if x != 0 else 0
+            )
+        else:
+            dc_link = network.links.index[network.links.carrier == "DC"]
+            link_widths = pd.Series(0, index=network.links.index)
+            link_widths.loc[dc_link] = 2
     elif line_colors == "q_flow_max":
-        title = "Maximmal reactive power flows"
-        label = "flow in Mvar"
+        title = "Maximum reactive power flows"
+        label = "flow in pu"
         line_colors = abs(
             network.lines_t.q0.abs().max() / (network.lines.s_nom)
         )
+        if ext_width is not False:
+            line_widths = 0.5 + (line_colors / ext_width)
+        link_colors = pd.Series(data=0, index=network.links.index)
+        plot_background_grid(network, ax)
+    elif line_colors == "dlr":
+        title = "Dynamic line rating"
+        label = "TWh above nominal capacity"
+        plot_background_grid(network, ax)
+        # Extract branch_capacity_factors
+        bcf_hv = self.args["branch_capacity_factor"]["HV"]
+        bcf_ehv = self.args["branch_capacity_factor"]["eHV"]
+        # calc min capacity per line in the given period
+        network.lines.s_max_pu = network.lines_t.s_max_pu.min()
+        network.lines.s_max_pu[network.lines.s_max_pu < 0.7] = 0.5
+        network.lines.s_max_pu[
+            (network.lines.s_max_pu > 0.5) & (network.lines.s_max_pu < 0.70)
+        ] = (bcf_hv + bcf_ehv) / 2
+        network.lines.s_max_pu[network.lines.s_max_pu >= 0.70] = 0.70
```

Comment:
Why are the lines from 2535 - 2539 are needed? I guess you needed to know the n-1 security factor per line to calculate the maximum "static" line loading. 
Since they are part of the s_max_pu time series you tried to extract them from the minimum s_max_ pu (because the minimum DLR factor is 1). 
But the following lines somehow overwrite this factor with the mean value, which is less accurate. 
Can you explain (and add to the code) why this is needed?

Reply:
The mentioned lines were removed.